File: lms-frontend/src/app/_model/books.ts
import { Author } from "./author";
import { Genre } from "./genre";

export class Books {
    bookId: number;
    authorId: number;
    genreId: number;
    bookName: string;
    author: Author;
    genre: Genre;
    authorName: string;
    genreName: string;
    noOfCopies: number;
}

File: lms-backend/src/main/java/com/ibizabroker/lms/entity/Books.java
package com.ibizabroker.lms.entity;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "book")
public class Books {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer bookId;
    String bookName;

    @ManyToOne
    @JoinColumn(name = "author_id", nullable = false)
    Author author;

    @ManyToOne
    @JoinColumn(name = "genre_id", nullable = false)
    Genre genre;

    Integer noOfCopies;

    public void borrowBook() {
        this.noOfCopies--;
    }

    public void returnBook() {
        this.noOfCopies++;
    }

    public Books() {
    }
}
